.github/workflows/apple.yml

```diff
@@ -40,135 +40,39 @@ jobs:
-            py: 'cpython-3.13'
-            options: 'pgo'
-          - target_triple: 'aarch64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'pgo+lto'
-
-          - target_triple: 'aarch64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'freethreaded+debug'
-          - target_triple: 'aarch64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'freethreaded+pgo'
-          - target_triple: 'aarch64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'freethreaded+pgo+lto'
-
-          # macOS on Intel hardware. This is pretty straightforward. We exclude
-          # noopt because it doesn't provide any compelling advantages over PGO
-          # or LTO builds.
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.9'
-            options: 'debug'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.9'
-            options: 'pgo'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.9'
-            options: 'pgo+lto'
-
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.10'
-            options: 'debug'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.10'
-            options: 'pgo'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.10'
-            options: 'pgo+lto'
-
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.11'
-            options: 'debug'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.11'
-            options: 'pgo'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.11'
-            options: 'pgo+lto'
-
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.12'
-            options: 'debug'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.12'
-            options: 'pgo'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.12'
-            options: 'pgo+lto'
-
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'debug'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'pgo'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'pgo+lto'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'freethreaded+debug'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'freethreaded+pgo'
-          - target_triple: 'x86_64-apple-darwin'
-            py: 'cpython-3.13'
-            options: 'freethreaded+pgo+lto'
     needs:
+      - generate-matrix
       - pythonbuild
-    runs-on: depot-macos-14
+    strategy:
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
```

Comment:
I did not realize this magic was possible. Otherwise I would have done this months ago.

Reply:
It's fairly new. I'm glad you're into it!